src/lib/Controller/api/system.c

```diff
@@ -159,18 +161,39 @@ const char *wbu_system_webots_tmp_path() {
             continue;
           if (s.st_mtime < most_recent)
             continue;
-          sscanf(entry->d_name, "webots-%d", &webots_pid);
+          sscanf(entry->d_name, "webots-%d%s", &webots_pid, random_part);
```

Comment:
The `random_part` buffer could overflow:
```suggestion
          if (strlen(entry->d_name) > 64) {
              continue; // avoid buffer overflow
          }
          sscanf(entry->d_name, "webots-%d%s", &webots_pid, random_part);
```

Reply:
Thank you for the pointer, our test suite just warn me about the same issue (https://travis-ci.com/github/cyberbotics/webots/jobs/322059423#L275)

Reply:
I would rather use:
```suggestion
          sscanf(entry->d_name, "webots-%d-%63s", &webots_pid, random_part);
```